espnet2/spk/loss/aamsoftmax.py

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/python
+# -*- encoding: utf-8 -*-
+# code from https://github.com/clovaai/voxceleb_trainer/blob/master/loss/aamsoftmax.py
+# Adapted from https://github.com/wujiyang/Face_Pytorch (Apache License)
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import time, pdb, numpy, math
+
+class LossFunction(nn.Module):
```

Comment:
It might be better to define a common abstract Loss class if you want to add more losses later.